src/scribe_data/utils.py

```diff
@@ -420,3 +445,84 @@ def check_and_return_command_line_args(
         python {all_args[0]} '["comma_separated_sets_in_quotes"]'
         """
     )
+
+
+def translation_interrupt_handler(source_language, translations):
+    """
+    Handles interrupt signals and saves the current translation progress.
+
+    Parameters
+    ----------
+        source_language : str
+            The source language being translated from.
+
+        translations : list[dict]
+            The current list of translations.
+    """
+    print("\nThe interrupt signal has been caught and the current progress is being saved...")
+    with open(f"{get_language_dir_path(source_language)}/formatted_data/translated_words.json", 'w', encoding='utf-8') as file:
+        json.dump(translations, file, ensure_ascii=False, indent=4)
+    print("The current progress is saved to the translated_words.json file.")
+    exit()
+
+def get_target_langcodes(source_lang)->list[str]:
+    """
+    Returns a list of target language ISO codes for translation.
+
+    Parameters
+    ----------
+        source_lang : str
+            The source language being translated from.
+
+    Returns
+    -------
+        list[str]
+            A list of target language ISO codes.
+    """
+    target_langcodes=[]
+    for lang in get_scribe_languages():
+        if lang!=source_lang:
+            target_langcodes.append(get_language_iso(lang))
+        else:
+            continue
+    return target_langcodes
+
+def translate_to_other_languages(source_language, word_list, translations, batch_size=10):
+    """
+    Translates a list of words from the source language to other target languages using batch processing.
+
+    Parameters
+    ----------
+        source_language : str
+            The source language being translated from.
+
+        word_list : list[str]
+            The list of words to translate.
+
+        translations : dict
+            The current dictionary of translations.
+
+        batch_size : int
+            The number of words to translate in each batch.
+    """
+    model = M2M100ForConditionalGeneration.from_pretrained("facebook/m2m100_418M")
+    tokenizer = M2M100Tokenizer.from_pretrained("facebook/m2m100_418M")
+
+    for i in range(0, len(word_list), batch_size):
+        batch_words = word_list[i:i+batch_size]
+        print(f"Translating batch {i//batch_size + 1}: {batch_words}")
+        for lang_code in get_target_langcodes(source_language):
+            tokenizer.src_lang = get_language_iso(source_language)
+            encoded_words = tokenizer(batch_words, return_tensors="pt", padding=True)
```

Comment:
Here, it tokenizes the batch of words and converts them into a format suitable for the model. Due to this it becomes relatively faster to translate words.